File: src/main/java/com/LilG/Com/pkmn/pokemon.java
package com.LilG.Com.pkmn;

import java.util.BitSet;

/**
 * Created by lil-g on 11/14/16.
 */
public interface pokemon {

    short getHealth();

    short getMaxHealth();

    String getName();

    Type getMainType();

    Type getSecondaryType();

    BitSet getStatus();

    boolean setStatus(StatusEffect status);

    boolean getStatus(StatusEffect status);

    int doDamage(int damageToDeal, Type type);

    int takeDamage(int damageTaken, Type type);

    Ability getAbility();
}

File: src/main/java/com/LilG/pkmn/pokemon.kt
package com.LilG.pkmn

import java.util.*

/**
 * Created by lil-g on 11/14/16.
 */
interface pokemon {

    val health: Short

    val maxHealth: Short

    val name: String

    val mainType: Type

    val secondaryType: Type

    val status: BitSet

    fun setStatus(status: StatusEffect): Boolean

    fun getStatus(status: StatusEffect): Boolean

    fun doDamage(damageToDeal: Int, type: Type): Int

    fun takeDamage(damageTaken: Int, type: Type): Int

    val ability: Ability
}